architecture/ambient/ztunnel.md

```diff
@@ -0,0 +1,227 @@
+* Otherwise, we forward the request to the destination using HBONE. If the destination is a Service, we resolve this to a specific pod IP.
+
+In all cases, we spoof the original source IP.
+
+### Inbound Passthrough
+
+Traffic entering a pod that is not transmitted over HBONE is handled by the "inbound passthrough" code path, on port 15006.
+
+This is fairly straightforward.
+
+First, we need to check that this traffic is allowed.
+Traffic may be denied by RBAC policies (especially from a `STRICT` mode enforcement, which denies plaintext traffic).
+
+If it is, we will forward to the target destination.
+
+#### Hairpin
+
+In the case that the destination has a waypoint, it must have been bypassed to reach this path.
+How we handle this is [under discussion](https://docs.google.com/document/d/1uM1c3zzoehiijh1ZpZuJ1-SzuVVupenv8r5yuCaFshs/edit#heading=h.dwbqvwmg6ud3).
+
+### Inbound
+
+Traffic entering a pod over HBONE will be handled by the "inbound" code path, on port 15008.
+
+Incoming requests have multiple "layers": TLS wrapping HTTP CONNECT wrapping the users connection.
+
+To unwrap the first layer, we terminate TLS.
+As part of this, we need to pick the correct certificate to serve.
+As discussed in [HBONE](#hbone), this is based on the destination IP.
+Additionally, we enforce the peer has a valid mesh identity (but do not assert _which_ identity, yet).
+
+Next, we terminate the CONNECT.
+From the [headers](#headers), we know the target destination.
+If the target destination has a waypoint, we enforce the request is coming from that waypoint. Otherwise, the request is rejected.
+If there is no waypoint, ztunnel will enforce RBAC policies against the request.
+
+If all checks pass, ztunnel will open a connection to the target. This will spoof the source IP (from `Forwarded` for waypoints, or the incoming IP otherwise).
+Once the connection is established we return a 200 HTTP code, and bi-directionally copy data to/from the tunnel to the destination.
+
+## Certificates
+
+Ztunnel certificates are based on the standard Istio SPIFFE format: `spiffe://<trust domain>/ns/<ns>/sa/<sa>`.
+
+However, the identies of the certificates will be of the actual user workloads, not Ztunnel's own identity.
+This means Ztunnel will have multiple distinct certificates at a time, one for each unique identity (service account) running on its node.
+
+When fetching certificates, ztunnel will authenticate to the CA with its own identity, but request the identity of another workload.
+Critically, the CA must enforce that the ztunnel has access to request that identity.
+Requests for identities not running on the node are rejected.
+This is critical to ensure that a compromised node does not compromise the entire mesh.
+
+Note: ztunnel authenticates to the CA with a Kubernetes Service Account JWT token, which encodes the pod information, which is what enables this.
```

Comment:
Is it worth adding that any Ambient-compliant CAs will need to support delegated identity as well (which I think is possible via CSR SAN extensions)

Reply:
yes